src/udev/udev-builtin-path_id.c

```diff
@@ -634,9 +634,46 @@ static int find_real_nvme_parent(sd_device *dev, sd_device **ret) {
         if (r < 0)
                 return r;
 
-        /* If the 'real parent' is (still) virtual, e.g. for nvmf disks, refuse to set ID_PATH. */
-        if (path_startswith(devpath, "/devices/virtual/"))
-                return -ENXIO;
+        /* If the 'real parent' is (still) virtual, e.g. for nvmf disks, set ID_PATH
+	 * and nvmf devices will be showed up in by-path */
+        if (path_startswith(devpath, "/devices/virtual/")) {
+		_cleanup_(sd_device_unrefp) sd_device *fc_parent = NULL;
+		_cleanup_free_ char *first = NULL, *second = NULL, *nvme_pn =NULL;
+		const char *addr, *host_traddr, *pn, *fc_pn;
+
+		r = sd_device_get_sysattr_value(nvme, "address", &addr);
+		if (r < 0)
+			return r;
+
+		r = extract_many_words(&addr, ",", 0, &first, &second, NULL);
+		if (r != 2)
+			return -EINVAL;
+
+		host_traddr = second;
+		r = extract_many_words(&host_traddr, "-", 0, &first, &second, NULL);
```

Comment:
you are leaking meory, as `first` and `second` are overriden, but still point to memory that needs to be freed.